File: backend-spring/src/main/java/fr/ubo/kanban/services/CommentaireService.java
package fr.ubo.kanban.services;

import fr.ubo.kanban.dtos.commentaire.CommentaireRequestDto;
import fr.ubo.kanban.dtos.commentaire.CommentaireResponseDto;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface CommentaireService {
    List<CommentaireResponseDto> findByIdTache(Long idTache);
    CommentaireResponseDto create(Long idTache, CommentaireRequestDto dto);
    void delete(String id);
    CommentaireResponseDto createWithFiles(Long idTache, CommentaireRequestDto dto,
                                           List<MultipartFile> fichiers);
}

File: frontend/src/services/commentaireService.js
import api from './api.js'

export const commentaireService = {
  getByTache(idTache)       { return api.get(`/taches/${idTache}/commentaires`) },
  create(idTache, dto)      { return api.post(`/taches/${idTache}/commentaires`, dto) },
  delete(id)                { return api.delete(`/commentaires/${id}`) },
}